first draft VARCHAR datatype doc

ljsinclair opened the pull request:
Documentation based on "storage types currently" listed in https://molecula.atlassian.net/wiki/spaces/EN/pages/1216020481/Data+Types

SQL runs successfully in FeatureBase Cloud as of 2024-02-14

Additional detail should be added by the reviewer if required

jrbrinlee1 reviewed (commented):
I don't think the length is technically unlimited. It's just some maximum value of bytes which I see set to `const MaxStringLength int64 = 1024 * 1024 * 256 // 256 Mb for now` in the code. Otherwise, looks good.

ljsinclair commented:
Fixed and ready to roll.
I did the 'max' value as a common/include file so it could be inserted into both string and varchar.
Plus I updated a warning for using string for _id in tables
And last, added varchar to data-types-home under the string heading

jrbrinlee1 reviewed (approved):
LGTM
